File: test/CodeCracker.Test/CatchEmptyTests.cs
using System.Threading.Tasks;
using TestHelper;
using Xunit;

namespace CodeCracker.Test
{
    public class CatchEmptyTests : CodeFixTest<CatchEmptyAnalyzer, CatchEmptyCodeFixProvider>
    {

        [Fact]
        public async Task CatchEmptyAnalyserCreateDiagnostic()
        {
            var source = @"
    using System;

    namespace ConsoleApplication1
    {
        class TypeName
        {
            public async Task Foo()
            {
                try
                {
                   // do something
                }
                catch (Exception ex)
                {
                   int x = 0;
                }
            }
        }
    }";

            await VerifyCSharpHasNoDiagnosticsAsync(source);
        }

        [Fact]
        public async Task WhenFindCatchEmptyThenPutExceptionClass()
        {
            var test = @"
    using System;

    namespace ConsoleApplication1
    {
        class TypeName
        {
            public async Task Foo()
            {
                try
                {
                   // do something
                }
                catch
                {
                   int x = 0;
                }
            }
        }
    }";

            var fixtest = @"
    using System;

    namespace ConsoleApplication1
    {
        class TypeName
        {
            public async Task Foo()
            {
                try
                {
                   // do something
                }
                catch (Exception ex)
                {
                   int x = 0;
                }
            }
        }
    }";
            await VerifyCSharpFixAsync(test, fixtest, 0);
        }
    }
}

File: test/VisualBasic/CodeCracker.Test/Design/CatchEmptyTests.vb
Imports CodeCracker.Test.TestHelper
Imports Xunit

Public Class CatchEmptyTests
    Inherits CodeFixTest(Of CatchEmptyAnalyzer, CatchEmptyCodeFixProvider)

    <Fact>
    Public Async Function CatchEmptyAnalyserCreateDiagnostic() As Task
        Const source = "
Imports System
Namespace ConsoleApplication1
    Class TypeName
        Public Async Function Foo() As Task
            Try
                ' Do something
            Catch ex as Exception
                Dim x = 0
            End Try
        End Function
    End Class
End Namespace"
        Await VerifyBasicHasNoDiagnosticsAsync(source)
    End Function

    <Fact>
    Public Async Function WhenFindCatchEmptyThenPutExceptionClass() As Task
        Const source = "
Imports System
Namespace ConsoleApplication1
    Class TypeName
        Public Async Function Foo() As Task
            Try
                ' Do something
            Catch
                Dim x = 0
            End Try
        End Function
    End Class
End Namespace"

        Const fix = "
Imports System
Namespace ConsoleApplication1
    Class TypeName
        Public Async Function Foo() As Task
            Try
                ' Do something
            Catch ex As Exception
                Dim x = 0
            End Try
        End Function
    End Class
End Namespace"
        Await VerifyBasicFixAsync(source, fix, 0)
    End Function

End Class

